code/debris/debris.cpp

```diff
@@ -79,6 +79,42 @@ static void debris_start_death_roll(object *debris_obj, debris *debris_p)
     debris_obj->flags.set(Object::Object_Flags::Should_be_dead);
 }
 
+int clearDebrisAction(sexp::SEXPParameterExtractor* extractor) {
+	bool hullDebris = extractor->getOptionalArg(true);
+	bool nonHullDebris = extractor->getOptionalArg(true);
+
+	for (auto& debris_entry : Debris) {
+		if (!(debris_entry.flags[Debris_Flags::Used]))
+			continue;
+
+		if (debris_entry.is_hull) {
+			if (hullDebris) {
+				debris_start_death_roll(&Objects[debris_entry.objnum], &debris_entry);
+			}
+		} else {
+			if (nonHullDebris) {
+				debris_start_death_roll(&Objects[debris_entry.objnum], &debris_entry);
```

Comment:
If this sexp is supposed to behave just like the `clear-weapons` sexp, then don't call `debris_start_death_roll`.  Instead just set the `Should_be_dead` object flag.

Reply:
`debris_start_death_roll` has the advantage that it takes care of sending the relevant multi packet so I chose that for "automatic" multi support.

Reply:
Yeah, but that function also creates a fireball and plays an explosion sound, two things you don't want to happen when you're just deleting all debris.

If I were writing this sexp I would just overload the functions that implement `remove-weapons`; the process is identical except for the `weapon_info_index` check.  Come to think of it, why not change this PR into a pure external sexp system, and I'll create a separate PR to add the `clear-debris` sexp?

Reply:
I'm fine with not having that sexp and removed it from the PR.

Reply:
Ok.